src/dune/dune_file.mli

```diff
@@ -55,15 +55,15 @@ module Buildable : sig
 end
 
 module Public_lib : sig
-  type t =
-    { name : Loc.t * Lib_name.t  (** Full public name *)
-    ; package : Package.t  (** Package it is part of *)
-    ; sub_dir : string option
-          (** Subdirectory inside the installation directory *)
-    }
+  type t
 
-  val name : t -> Lib_name.t
+  (** Subdirectory inside the installation directory *)
+  val sub_dir : t -> string option
 
+  (** Full public name *)
+  val name : t -> Loc.t * Lib_name.t
```

Comment:
It seems the interface and the use-sites will become more straightforward if we provide two functions isntead:

```ocaml
val name : t -> Lib_name.t
val loc : t -> Loc.t
```